roqoqo/src/devices/square_lattice.rs

```diff
@@ -0,0 +1,452 @@
+                .insert(gate.to_string(), gatetimes);
+        }
+        self
+    }
+
+    /// Function that allows to set a unifromg gate time per gate type for the two-qubit-gates.
+    ///
+    /// # Arguments
+    ///
+    /// * `gate` - hqslang name of the two-qubit-gate.
+    /// * `gate_time` - gate time for the given gate type, valid for all qubits in the device.
+    ///
+    /// # Returns
+    ///
+    /// An SquareLatticeDevice with updated gate times.
+    ///
+    pub fn set_all_two_qubit_gate_times(mut self, gate: &str, gate_time: f64) -> Self {
+        if let Some(gate_times) = self.generic_device.two_qubit_gates.get_mut(gate) {
+            for (_, gatetime) in gate_times.iter_mut() {
+                *gatetime = gate_time
+            }
+        } else {
+            let mut gatetimes: HashMap<(usize, usize), f64> =
+                HashMap::with_capacity(self.number_qubits() * 4);
+            // insert horizontal terms
+            for row in 0..self.number_rows() {
+                for column in 0..self.number_columns() - 1 {
+                    gatetimes.insert(
+                        (
+                            row * self.number_columns + column,
+                            row * self.number_columns + column + 1,
+                        ),
+                        gate_time,
+                    );
+                    gatetimes.insert(
+                        (
+                            row * self.number_columns + column + 1,
+                            row * self.number_columns + column,
+                        ),
+                        gate_time,
+                    );
+                }
+            }
+            // insert vertical terms
+            for row in 0..self.number_rows - 1 {
+                for column in 0..self.number_columns {
+                    gatetimes.insert(
+                        (
+                            row * self.number_columns + column,
+                            (row + 1) * self.number_columns + column,
+                        ),
+                        gate_time,
+                    );
+                    gatetimes.insert(
+                        (
+                            (row + 1) * self.number_columns + column,
+                            row * self.number_columns + column,
+                        ),
+                        gate_time,
+                    );
+                }
+            }
+            self.generic_device
+                .two_qubit_gates
+                .insert(gate.to_string(), gatetimes);
+        }
+        self
+    }
+
+    /// Setting the gate time of a single qubit gate.
+    ///
+    /// # Arguments
+    ///
+    /// * `gate` - hqslang name of the single-qubit-gate.
+    /// * `qubit` - The qubit for which the gate time is set
+    /// * `gate_time` - gate time for the given gate.
+    ///
+    /// # Returns
+    ///
+    /// An SquareLatticeDevice with updated gate times or
+    ///
```

Comment:
This one and the next one have unfinished sentences

Reply:
This one needs no raises, because it can not fail (the PyResult is just a necessary PyO3 signature for new)